process/sync/metablock.go

```diff
@@ -405,6 +405,8 @@ func (boot *MetaBootstrap) requestMiniBlocksFromHeaderWithNonceIfMissing(_ uint3
 		return
 	}
 
+	boot.requestedItemsHandler.Sweep()
+
 	hashes := make([][]byte, len(header.MiniBlockHeaders))
```

Comment:
You forgot to call Has from requestedItemsHandler in next for. You should replace lines between 410-413 with these:

	hashes := make([][]byte, 0)
	for i := 0; i < len(header.MiniBlockHeaders); i++ {
		if boot.requestedItemsHandler.Has(string(header.MiniBlockHeaders[i].Hash)) {
			continue
		}

		hashes = append(hashes, header.MiniBlockHeaders[i].Hash)
	}

Reply:
done.